crates/viewer/re_dataframe_ui/src/datafusion_table_widget.rs

```diff
@@ -695,33 +724,75 @@ impl egui_table::TableDelegate for DataFusionTableDelegate<'_> {
+            Self::with_row_batch(self.display_record_batches, cell.row_nr as usize)
+        {
+            let column = &display_record_batch.columns()[col_index];
 
-                // TODO(#9029): it is _very_ unfortunate that we must provide a fake timeline, but
-                // avoiding doing so needs significant refactoring work.
-                column.data_ui(
-                    self.ctx,
-                    ui,
-                    &re_viewer_context::external::re_chunk_store::LatestAtQuery::latest(
-                        TimelineName::new("unknown"),
-                    ),
-                    row_index,
-                    None,
-                );
+            // TODO(#9029): it is _very_ unfortunate that we must provide a fake timeline, but
+            // avoiding doing so needs significant refactoring work.
+            column.data_ui(
+                self.ctx,
+                ui,
+                &re_viewer_context::external::re_chunk_store::LatestAtQuery::latest(
+                    TimelineName::new("unknown"),
+                ),
+                batch_index,
+                None,
+            );
+        }
+    }
 
-                break;
-            } else {
-                row_index -= row_count;
-            }
+    fn row_ui(&mut self, ui: &mut Ui, _row_nr: u64) {
+        let has_context_menu = self.blueprint.partition_links.is_some();
+        if has_context_menu {
+            ui.response().container_context_menu(|ui| {
+                let selection = TableSelectionState::load(ui.ctx(), self.session_id);
+
+                // re_table will ensure that the right-clicked row is always selected.
+                let selected_items = selection.selected_rows;
+
+                if let Some(partition_links_spec) = &self.blueprint.partition_links {
+                    let label = format!(
+                        "Open {} partition{}",
+                        selected_items.len(),
+                        format_plural_s(selected_items.len())
+                    );
+                    if ui
+                        .add(icons::OPEN_RECORDING.as_button_with_label(ui.tokens(), label))
+                        .clicked()
+                    {
+                        let column_index = self.columns.iter().position(|col| {
+                            col.blueprint.display_name.as_ref()
+                                == Some(&partition_links_spec.column_name)
+                        });
+
+                        // Let's open the tabs in order
+                        for row in selected_items.iter().copied().sorted() {
+                            if let Some((display_record_batch, batch_index)) =
+                                Self::with_row_batch(self.display_record_batches, row as usize)
+                            {
+                                if let Some(column) = column_index
+                                    .and_then(|col| display_record_batch.columns().get(col))
```

Comment:
Quite a lot of indentation happening here, could at least combine these if statements